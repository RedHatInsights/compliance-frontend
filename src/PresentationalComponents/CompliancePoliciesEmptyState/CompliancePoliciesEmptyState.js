import React from 'react';
import {
    Title,
    TextContent,
    Button,
    Bullseye,
    EmptyState,
    EmptyStateBody,
    EmptyStateSecondaryActions,
    EmptyStateIcon
} from '@patternfly/react-core';
import { CloudServerIcon } from '@patternfly/react-icons';
import CreatePolicy from '../../SmartComponents/CreatePolicy/CreatePolicy';
import propTypes from 'prop-types';

const CompliancePoliciesEmptyState = ({ title }) => (
    <Bullseye>
        <EmptyState>
            <EmptyStateIcon style={{ fontWeight: '500', color: 'var(--pf-global--primary-color--100)' }}
                size="xl" title="Compliance" icon={CloudServerIcon} />
            <br/>
            <Title size="lg">{ title }</Title>
            <br/>
            <EmptyStateBody>
                <TextContent>
                    The Compliance service uses SCAP policies to track your organization&#39;s adherence to
                    compliance requirements.
                </TextContent>
                <TextContent>
                    Get started by adding a policy, or read documentation about how to connect OpenSCAP to the
                    Compliance service.
                </TextContent>
            </EmptyStateBody>
            <CreatePolicy onWizardFinish={() => { location.reload(); }} />
            <EmptyStateSecondaryActions>
                <Button variant='link' component='a' target='_blank' rel='noopener noreferrer'
                    href='https://www.open-scap.org/getting-started/' >
                    Learn about OpenSCAP and Compliance
                </Button>
            </EmptyStateSecondaryActions>
        </EmptyState>
    </Bullseye>
);

CompliancePoliciesEmptyState.propTypes = {
    title: propTypes.string
};

CompliancePoliciesEmptyState.defaultProps = {
    title: 'No policies'
};

export default CompliancePoliciesEmptyState;
