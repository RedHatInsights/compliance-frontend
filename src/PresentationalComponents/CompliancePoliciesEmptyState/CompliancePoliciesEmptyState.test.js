import toJson from 'enzyme-to-json';

import CompliancePoliciesEmptyState from './CompliancePoliciesEmptyState';
jest.mock('@redhat-cloud-services/frontend-components-inventory-compliance', () => {
    const ComplianceRemediationButton = () => <button>Remediations</button>;
    return ComplianceRemediationButton;
});

describe('CompliancePoliciesEmptyState', () => {
    it('expect to render without error', () => {
        const wrapper = shallow(
            <CompliancePoliciesEmptyState />
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
