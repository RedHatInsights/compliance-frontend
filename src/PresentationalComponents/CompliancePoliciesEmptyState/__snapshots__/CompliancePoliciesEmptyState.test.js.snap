// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompliancePoliciesEmptyState expect to render without error 1`] = `
<Bullseye>
  <EmptyState>
    <EmptyStateIcon
      icon={[Function]}
      size="xl"
      style={
        Object {
          "color": "var(--pf-global--primary-color--100)",
          "fontWeight": "500",
        }
      }
      title="Compliance"
    />
    <br />
    <Title
      size="lg"
    >
      No policies
    </Title>
    <br />
    <EmptyStateBody>
      <TextContent>
        The Compliance service uses SCAP policies to track your organization's adherence to compliance requirements.
      </TextContent>
      <TextContent>
        Get started by adding a policy, or read documentation about how to connect OpenSCAP to the Compliance service.
      </TextContent>
    </EmptyStateBody>
    <Connect(CreatePolicy)
      onWizardFinish={[Function]}
    />
    <EmptyStateSecondaryActions>
      <Component
        component="a"
        href="https://www.open-scap.org/getting-started/"
        rel="noopener noreferrer"
        target="_blank"
        variant="link"
      >
        Learn about OpenSCAP and Compliance
      </Component>
    </EmptyStateSecondaryActions>
  </EmptyState>
</Bullseye>
`;
